### PR TITLE
Fetch patch from local gecko repo and apply to git repo and Stop installing `patch` in our Docker image

### PR DESCRIPTION
**Pull request by PromiseFru**
Closes: #1506

### CODE REVIEW

File: scripts/commit_classifier.py [hunk 1]
```diff
@@ -392,29 +392,14 @@ def load_user(phid):
...
-                        "-c",
-                        f"user.email={author_email}",
-                        "commit",
-                        "-am",
-                        message,
-                    ],
-                    check=True,
-                    cwd=self.git_repo_dir,
-                )
+        latest_rev = repository.get_revs(hg, f"-{len(stack)}")[-1]
```
Comment (suhaibmujahid) [inline scripts/commit_classifier.py:395]: Why do we need here to retrieve `len(stack)`? Why not just the last one (i.e., `-1`)?

**Review by suhaibmujahid — COMMENTED** [review]
Thank you, @PromiseFru! Did you test/execute the code to make sure that it works as expected?

**Comment by PromiseFru** [comment]
> Thank you, @PromiseFru! Did you test/execute the code to make sure that it works as expected?

Hello @suhaibmujahid, yes, I executed the code to ensure it fetches the patch from the local Mercurial repo, and it did that. I used the diff-id [190849](https://phabricator.services.mozilla.com/D190849) to test.

```bash
python scripts/commit_classifier.py testlabelselect ../gecko/ --git_repo_dir ../git_gecko/ --phabricator-deployment dev --diff-id 190849
```

[commit_classifier_output.txt](https://github.com/mozilla/bugbug/files/13064337/commit_classifier_output.txt)

File: scripts/commit_classifier.py (hunk 1 above)
Comment (PromiseFru) [reply scripts/commit_classifier.py]: Using the last commit (-1) is more efficient, as you suggested. I've made updates to the code to reflect this change.

**Review by suhaibmujahid — APPROVED** [review]
Thank you. LGTM!

**Review by marco-c — APPROVED** [review]

File: infra/dockerfile.commit_retrieval [hunk 2]
```diff
@@ -6,7 +6,7 @@ ENV PATH="${PATH}:/git-cinnabar"
 # libcurl4 is required by git-cinnabar.
 # patch is required by the commit classifier script.
```
Comment (marco-c) [inline infra/dockerfile.commit_retrieval:7]: @PromiseFru, I just realized this comment is no longer applicable, could you submit a follow-up PR to remove it?

Comment (PromiseFru) [reply infra/dockerfile.commit_retrieval:7]: Certainly! I'll get right on it.

**Pull request merged**